sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/AadRestOperationConfiguration.java

```diff
@@ -29,51 +33,77 @@
 public class AadRestOperationConfiguration {
```

Comment:
I suggest we split this class into two classes, `AadOAuth2ClientRestOperationConfiguration` and `AadResourceServerRestOperationConfiguration`.

Reply:
And on `AadOAuth2ClientConfiguration` import this first one, on `AadResourceServerConfiguration` import the other. I agree with what @moarychan said, when only either oauth2 client or resource server is used, we don't need both beans in the context.